<b>Conclusões a partir da análise dos dados</b><br> 
Questionamentos:<br> 
P1: Mais ou menos modelos estão usando fontes alternativas de combustível? Se mais, quanto mais?<br> 
P2: Quais são as características dos veículos SmartWay? Elas mudaram ao longo do tempo (mpg, classificação de gases)?<br> 

<b>Conclusions from data analysis</b><br>
Questions:<br>
Q1: Are more or fewer models using alternative fuel sources? If more, how much more?<br>
Q2: What are the characteristics of SmartWay vehicles? Have they changed over time (mpg, gas rating)?<br>


In [31]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline


df_08 = pd.read_csv('clean_08.csv')
df_18 = pd.read_csv('clean_18.csv')

P1: Mais ou menos modelos estão usando fontes alternativas de combustível? Se mais, quanto mais?<br> 
Q1: Are more or fewer models using alternative fuel sources? If more, how much more?

In [32]:
df_08['fuel'].unique()

array(['Gasoline', 'CNG', 'ethanol', 'gas'], dtype=object)

In [33]:
df_18['fuel'].unique()

array(['Gasoline', 'Diesel', 'Ethanol', 'Electricity', 'Gas'],
      dtype=object)

<b>Analisando os tipos de combustíveis (Analyzing fuel types)</b><br> 

O GNC - Gás Natural Comprimido é extraído de matérias orgânicas, compostagens e bagaços, este modelo é mais seguro quando comparado ao GNV, por utilizar as matérias orgânicas o seu gás é mais leve e é evaporado rapidamente pela atmosfera o que evita riscos de incêndio e explosão.<br> 
Fonte: https://transac.com.br/noticia/4/transporte-de-combustiveis/novidades-em-combustivel-conheca-o-gnc

CNG - Compressed Natural Gas is extracted from organic materials, compost and bagasse. This model is safer when compared to VNG, as it uses organic materials, its gas is lighter and evaporates quickly into the atmosphere, which prevents the risk of fire and explosion.<br>
Source:https://transac.com.br/noticia/4/transporte-de-combustiveis/novidades-em-combustivel-conheca-o-gnc

Tido como o combustível mais limpo do mundo, o etanol pode reduzir as emissões de gás carbônico de ponta a ponta (da produção até o uso nos veículos) em até 90% em relação à gasolina e, por ser feito de cana-de-açúcar ou de milho, é renovável, diferentemente do diesel e da gasolina.
Fonte: https://exame.com/negocios/vantajoso-abastecer-etanol/

Considered the cleanest fuel in the world, ethanol can reduce carbon dioxide emissions from end to end (from production to use in vehicles) by up to 90% compared to gasoline and, as it is made from sugarcane or corn and it is renewable, unlike diesel and gasoline.
Source: https://exame.com/negocios/vantajoso-abastecer-etanol/

In [34]:
fuel_08 = df_08['fuel'].unique()
fuel_18 = df_18['fuel'].unique()

In [35]:
def calc_fuel_prop(df):
    
    fuels = df['fuel'].unique()
    for fuel in fuels:
        total = df.shape[0]
        print('Combustível: {0} Quantidade: {1}'.format(
                fuel, df.query("fuel == @fuel")['fuel'].count()/total))

print("08")
calc_fuel_prop(df_08)
print("18")
calc_fuel_prop(df_18)

08
Combustível: Gasoline Quantidade: 0.9969604863221885
Combustível: CNG Quantidade: 0.0010131712259371835
Combustível: ethanol Quantidade: 0.0010131712259371835
Combustível: gas Quantidade: 0.0010131712259371835
18
Combustível: Gasoline Quantidade: 0.9002403846153846
Combustível: Diesel Quantidade: 0.02283653846153846
Combustível: Ethanol Quantidade: 0.03125
Combustível: Electricity Quantidade: 0.014423076923076924
Combustível: Gas Quantidade: 0.03125


P1: Mais ou menos modelos estão usando fontes alternativas de combustível? Se mais, quanto mais?<br> 
Q1: Are more or fewer models using alternative fuel sources? If more, how much more?

R1: Comparando os dados de percentual de modelos de veículos que usam cada tipo de combustível, pode-se observar uma queda de 10,74% na quantidade de veículos que usam gasolina, assim como o aumento na quantidade de veículos que usam fontes alternativas de combustível como Etanol (aumento de 3,2%) e gás natural (aumento de 3,2%).</b> 

A1: Comparing the data on the percentage of vehicle models that use each type of fuel, we can see a 10.74% drop in the number of vehicles that use gasoline, as well as an increase in the number of vehicles that use alternative fuel sources such as Ethanol (3.2% increase) and natural gas (3.2% increase).</b>

P2: Quais são as características dos veículos SmartWay? Elas mudaram ao longo do tempo (mpg, classificação de gases)?<br> 
Q2: What are the characteristics of SmartWay vehicles? Have they changed over time (mpg, gas rating)?

In [36]:
df_08.groupby("smartway").describe()

displ                                                   cyl  \
          count      mean       std  min    25%  50%  75%  max  count   
smartway                                                                
no        607.0  4.056837  1.203933  1.3  3.200  3.8  4.9  8.4  607.0   
yes       380.0  2.602895  0.623436  1.3  2.275  2.4  3.0  5.0  380.0   

                    ... cmb_mpg       greenhouse_gas_score            \
              mean  ...     75%   max                count      mean   
smartway            ...                                                
no        6.771005  ...    19.0  23.0                607.0  4.149918   
yes       4.826316  ...    25.0  46.0                380.0  6.868421   

                                              
               std  min  25%  50%  75%   max  
smartway                                      
no        1.629634  0.0  3.0  5.0  5.0   7.0  
yes       0.827338  6.0  6.0  7.0  7.0  10.0  

[2 rows x 56 columns]

In [37]:
df_18.groupby("smartway").describe()

displ                                                 cyl            \
          count      mean       std  min  25%  50%  75%  max  count      mean   
smartway                                                                        
Elite      17.0  1.970588  0.277859  1.5  2.0  2.0  2.0  2.5   17.0  4.000000   
No        724.0  3.270304  1.346987  1.4  2.0  3.0  3.9  8.0  724.0  5.720994   
Yes        91.0  1.753846  0.420399  1.2  1.5  1.6  2.0  3.5   91.0  3.923077   

          ... cmb_mpg        greenhouse_gas_score                             \
          ...     75%    max                count       mean       std   min   
smartway  ...                                                                  
Elite     ...    67.0  106.0                 17.0  10.000000  0.000000  10.0   
No        ...    25.0   60.0                724.0   4.276243  1.256106   1.0   
Yes       ...    34.0  106.0                 91.0   7.538462  0.860332   7.0   

                                  
           25%   50%   75%   max  
smartway                          
Elite     10.0  10.0  10.0  10.0  
No         3.0   4.0   5.0  10.0  
Yes        7.0   7.0   8.0  10.0  

[3 rows x 56 columns]

R2.Comparando os dados de veículos SmartWay de 2008 e 2018 percebe-se uma melhora na economia de combustível (aumento no mpg de 46 para 106), assim como a melhora na classificação de emissão de gases do efeito estufa de 2008 para 2018 (aumento da média de 6.87 para 7.54). </b> 

A2.Comparing the data from SmartWay vehicles from 2008 and 2018, we can see an improvement in fuel economy (increase in mpg from 46 to 106), as well as an improvement in the greenhouse gas emission rating from 2008 to 2018 (increase in the average from 6.87 to 7.54).</b>